File: GDKit/Classes/GDView/UIView+GDFrame.m
//
//  UIView+GDFrame.m
//  GDKit
//
//  Created by GDKit on 01/11/2022.
//  Copyright (c) 2022 GDKit. All rights reserved.
//

#import "UIView+GDFrame.h"

@implementation UIView (GDFrame)

- (CGPoint)cOrigin {
    
    return self.frame.origin;
}

- (void)setCOrigin:(CGPoint)cOrigin {
    
    CGRect frame = self.frame;
    frame.origin = cOrigin;
    self.frame = frame;
}

- (CGSize)cSize {
    
    return self.frame.size;
}

- (void)setCSize:(CGSize)cSize {
    
    CGRect frame = self.frame;
    frame.size = cSize;
    self.frame = frame;
}

- (CGFloat)height {
    
    return CGRectGetHeight(self.frame);
}

- (void)setHeight:(CGFloat)height {
    
    CGRect frame = self.frame;
    frame.size.height = height;
    self.frame = frame;
}

- (CGFloat)width {
    
    return CGRectGetWidth(self.frame);
}

- (void)setWidth:(CGFloat)width {
    
    CGRect frame = self.frame;
    frame.size.width = width;
    self.frame = frame;
}

- (CGFloat)top {
    
    return CGRectGetMinY(self.frame);
}

- (void)setTop:(CGFloat)top {
    
    CGRect frame = self.frame;
    frame.origin.y = top;
    self.frame = frame;
}

- (CGFloat)left {
    
    return CGRectGetMinX(self.frame);
}

- (void)setLeft:(CGFloat)left {
    
    CGRect frame = self.frame;
    frame.origin.x = left;
    self.frame = frame;
}

- (CGFloat)bottom {
    
    return CGRectGetMaxY(self.frame);
}

- (void)setBottom:(CGFloat)bottom {
    
    CGRect frame = self.frame;
    frame.origin.y = bottom - CGRectGetHeight(self.frame);
    self.frame = frame;
}

- (CGFloat)right {
    
    return CGRectGetMaxX(self.frame);
}

- (void)setRight:(CGFloat)right {
    
    CGRect frame = self.frame;
    frame.origin.x = right - CGRectGetWidth(self.frame);
    self.frame = frame;
}

@end

File: GDKit/Classes/GDView/UIView+GDFrame.swift
//
//  UIView+GDFrame.swift
//  GDKit
//
//  Created by SJ on 2022/1/15.
//

import Foundation

extension UIView {
    //let cOrigin: CGPoint //!< 位置
    //let cSize: CGSize //!< 大小
    
    open var height: CGFloat {
        get {
            return self.frame.height
        }
        set {
            var frame = self.frame
            frame.size.height = newValue
            self.frame = frame
        }
    } //!< 高度
    
    open var width: CGFloat {
        get {
            return self.frame.width
        }
        set {
            var frame = self.frame
            frame.size.width = newValue
            self.frame = frame
        }
    } //!< 宽度
    
    open var top: CGFloat {
        get {
            return self.frame.minY
        }
        set {
            var frame = self.frame
            frame.origin.y = newValue
            self.frame = frame
        }
    } //!< 上
    
    open var left: CGFloat {
        get {
            return self.frame.minX
        }
        set {
            var frame = self.frame
            frame.origin.x = newValue
            self.frame = frame
        }
    } //!< 左
    
    open var bottom: CGFloat {
        get {
            return self.frame.maxY
        }
        set {
            var frame = self.frame
            frame.origin.y = newValue - self.frame.height
            self.frame = frame
        }
    } //!< 下
    
    open var right: CGFloat {
        get {
            return self.frame.maxX
        }
        set {
            var frame = self.frame;
            frame.origin.x = newValue - self.frame.width
            self.frame = frame
        }
    } //!< 右
    
}
